analyzer/codechecker_analyzer/arg.py

```diff
@@ -9,6 +9,26 @@
 
 
 import argparse
+import sys
+
+from codechecker_common import logger
+
+
+LOG = logger.get_logger('system')
+
+
+class CtuAstModeAction(argparse.Action):
+    """
+    Asserts the presence of ctu mode in the arguments. It is not
+    meaningful to supply --ctu-ast-mode if ctu mode is not enabled.
+    """
+    def __call__(self, parser, namespace, values,
+                 option_string=None):
+        if 'ctu_phases' not in namespace:
+            LOG.error("Analyzer option 'ctu-ast-mode' requires CTU mode "
```

Comment:
I tried to run the following command but I can see this error message even I provided the --ctu option:
```sh
$ CodeChecker analyze ./build.json -o ./reports --ctu-ast-mode parse-on-demand --ctu
Analyzer option 'ctu-ast-mode' requires CTU mode enabled!
```

What is strange that if I insert the --ctu option before the --ctu-ast-mode I don't get this warning:
```sh
$ CodeChecker analyze ./build.json -o ./reports --ctu --ctu-ast-mode parse-on-demand
```

We should handle both cases correctly.

Reply:
OK, then I think I will revert this change. It seemed nice to have the check inside the argument action, but this behaviour shows that the actions are really run in the same order they were specified on the actual command-line invocation, and not as they are registered in the arg-parsing code. Thanks!